File: examples/python/mcp_tools_list.py
import os
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[2] / "sdks" / "python"))

from solo_client import SoloClient  # noqa: E402

solo = SoloClient(
    base_url=os.getenv("SOLO_URL", "http://127.0.0.1:17821"),
    bearer_token=os.getenv("SOLO_BEARER_TOKEN"),
)

session = solo.mcp_connect("solo-sdk-example", "0.0.0")

for tool in solo.mcp_list_tools(session):
    print(tool["name"])

File: examples/typescript/mcp-tools-list.ts
import { SoloClient } from "../../sdks/typescript/solo-client.js";

const solo = new SoloClient({
  baseUrl: process.env.SOLO_URL ?? "http://127.0.0.1:17821",
  bearerToken: process.env.SOLO_BEARER_TOKEN,
});

const session = await solo.mcpConnect({
  name: "solo-sdk-example",
  version: "0.0.0",
});

const tools = await solo.mcpListTools(session);
for (const tool of tools) {
  console.log(tool.name);
}
